sklearn/tests/test_isotonic.py

```diff
@@ -676,3 +676,22 @@ def test_isotonic_2darray_more_than_1_feature():
 
     with pytest.raises(ValueError, match=msg):
         iso_reg.transform(X_2d)
+
+
+def test_isotonic_regression_sample_weight_not_overwritten():
+    """Check that calling fitting function of isotonic regression will not
+    overwrite `sample_weight`.
+    Non-regression test for:
+    https://github.com/scikit-learn/scikit-learn/issues/20508
+    """
+    from sklearn.datasets import make_regression
```

Comment:
The import should be at the top of the file

Reply:
Not sure it's even needed. We can generate X,y like in the other tests. Something like
y = np.array([3, 7, 5, 9, 8, 7, 10])
x = np.arange(len(y))

Reply:
ups. my original test was changing the first value.